torch/testing/_internal/opinfo/definitions/nested.py

```diff
@@ -417,6 +439,45 @@ def sample_inputs_masked_select(
         )
 
 
+def sample_inputs_index_put(
+    op_info, device, dtype, requires_grad, op_kwargs=None, **kwargs
+):
+    for njt in _sample_njts(
+        device=device, dtype=dtype, requires_grad=requires_grad, dims=[2, 3, 4]
+    ):
+        for dim in range(njt.dim()):
+            indices = [
+                torch.tensor(list(range(njt.size(0))), device=njt.device),
+                *[
+                    torch.tensor([0] * njt.size(0), device=njt.device)
+                    for _ in range(dim - 1)
+                ],
+            ]
+            yield SampleInput(
+                njt,
```

Comment:
```suggestion
                njt.clone().detach(),
```

this is needed so in-place op runs don't bleed into each other

Reply:
done